Source/NSURL.m

```diff
@@ -2454,7 +2461,7 @@ - (void) setURL: (NSURL *)url
   [self setUser: [url user]];
   [self setPassword: [url password]];
   [self setPath: [url path]];
-  [self setQuery: [url query]];
+  [self setPercentEncodedQuery:[url query]];
```

Comment:
NSURL's query part can contain only characters from NSURLQueryCharacterSet, all other characters should be percent-encoded. Setting query in initializer with -setQuery leads name and value of NSURLQueryItem stay percent-encoded, and query property returns twice percent-encoded string.